scripts/play.ts

```diff
@@ -0,0 +1,550 @@
+
+  // Update scripts to support the monorepo and the given stage.
+  if (options.stage === "prod") {
+    await updatePackageJSON(
+      dir,
+      { scripts: { slicemachine: START_SLICEMACHINE_SCRIPT } },
+      { dryRun: options.dryRun },
+    );
+  } else {
+    await exec("yarn", ["add", "--dev", "cross-env"], {
+      cwd: dir,
+      stdio: "inherit",
+      dryRun: options.dryRun,
+    });
+
+    await updatePackageJSON(
+      dir,
+      {
+        scripts: {
+          slicemachine: `cross-env SM_ENV=${options.stage} ${START_SLICEMACHINE_SCRIPT}`,
+        },
+      },
+      { dryRun: options.dryRun },
+    );
+  }
+
+  // Update Slice Machine configuration with the correct API endpoint (if needed).
+  if (options.stage === "staging") {
+    await updateSliceMachineConfig(
+      dir,
+      { apiEndpoint: `https://${name}.cdn.wroom.io/api/v2` },
+      { dryRun: options.dryRun },
+    );
+  } else if (options.stage === "development") {
+    const apiEndpoint = new URL(
+      "./api/v2",
+      process.env.wroom_endpoint ?? "https://cdn.wroom.io",
+    );
+    apiEndpoint.hostname = `${name}.${apiEndpoint.hostname}`;
+
+    await updateSliceMachineConfig(
+      dir,
+      { apiEndpoint: apiEndpoint.toString() },
+      { dryRun: options.dryRun },
+    );
+  }
+
+  await exec(
+    "npx",
+    [
+      "@slicemachine/init@latest",
+      `--repository="${name}"`,
+      "--no-start-slicemachine",
+    ],
+    {
+      cwd: dir,
+      env: {
+        SM_ENV: options.stage === "prod" ? undefined : options.stage,
+      },
+      stdio: "inherit",
+      dryRun: options.dryRun,
+    },
+  );
+
+  // The Nuxt adapter edits the `endpoint` option as part of `@slicemachine/init`.
+  // We need to change it to the API endpoint to support different stages.
+  if (options.framework === "nuxt") {
+    const sliceMachineConfig: SliceMachineConfig = JSON.parse(
+      await fs.readFile(new URL("./slicemachine.config.json", dir), "utf8"),
+    );
+
+    if (sliceMachineConfig.apiEndpoint) {
+      const nuxtConfigPath = new URL("./nuxt.config.ts", dir);
+      const nuxtConfig = await fs.readFile(nuxtConfigPath, "utf8");
+      const newNuxtConfig = nuxtConfig.replace(
+        /endpoint: '.*',/,
+        `endpoint: '${sliceMachineConfig.apiEndpoint}',`,
+      );
+      await fs.writeFile(nuxtConfigPath, newNuxtConfig);
+    }
+  }
+}
+
+/**
+ * Clones a Git repository.
+ *
+ * @param url - The Git repository's URL.
+ * @param dir - The directory in which the repository is cloned.
+ * @param options - Options that determine the function's behavior.
+ */
+async function cloneGitRepo(
+  url: string,
+  dir: URL,
+  options: DryRunOption = {},
+): Promise<void> {
+  await exec("git", ["clone", "--depth=1", url, fileURLToPath(dir)], {
+    stdio: "inherit",
+    dryRun: options.dryRun,
+  });
+}
+
+/**
+ * Returns a random project name. The name is not guaranteed to be unique. Use
+ * `appendRandomHash` to virtually guarantee uniqueness.
```

Comment:
Thanks, changed to this description:

```
/**
 * Returns a random project name. A random 4-character hash is appended to the
 * name to prevent name collisions.
 */
```